addons/udes_stock_refactoring/models/stock_picking_type.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+U_STOCK_REFACTORING_CORE_LIFECYCLE_ACTIONS = [
+    ("group_by_move_line_key", "Group by Move Line Key"),
+    ("group_by_move_key", "Group by Move Key"),
+]
+
+
+class StockPickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    u_auto_unlink_empty = fields.Boolean(
+        string="Auto Unlink Empty",
+        default=True,
+        help="""
+        Flag to indicate whether to unlink empty pickings when searching for any empty picking in 
+        the system.
+        """,
+    )
+
+    # Picking lifecycle actions
+
+    u_move_line_key_format = fields.Char(
+        "Move Line Grouping Key",
+        help="A field name on stock.move.line that can be used to group move lines.",
+    )
+
+    u_move_key_format = fields.Char(
+        "Move Grouping Key", help="A field name on stock.move that can be to group move.",
+    )
+
+    u_post_confirm_action = fields.Selection(
+        selection=[
+            ("group_by_move_key", "Group by Move Key"),
+            ("batch_pickings_by_date", "Batch pickings by date"),
+            ("batch_pickings_by_date_priority", "Batch pickings by date and priority"),
+        ],
+        string="Post Confirm Action",
+        help="Choose the action to be taken after confirming a picking.",
+    )
+
+    u_post_assign_action = fields.Selection(
+        selection=U_STOCK_REFACTORING_CORE_LIFECYCLE_ACTIONS,
+        string="Post Assign Action",
+        help="Choose the action to be taken after reserving a picking.",
+    )
+
+    u_post_validate_action = fields.Selection(
+        selection=U_STOCK_REFACTORING_CORE_LIFECYCLE_ACTIONS,
```

Comment:
Do we need a U in front of the constant?

Reply:
I only added it on the very unlikely chance that Odoo put in their own refactor functionality and also used the `STOCK_REFACTORING_CORE_LIFECYCLE_ACTIONS` constant variable. I'm probably being too paranoid so happy to remove the U if you think its a bit over the top. :)

Reply:
This constant has now been removed as refactor actions are now in their own classes.